src/include/mpir_hwtopo.h

```diff
@@ -0,0 +1,114 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef MPIR_HWTOPO_H_INCLUDED
+#define MPIR_HWTOPO_H_INCLUDED
+
+/* hwtopo types */
+typedef enum {
+    MPIR_HWTOPO_TYPE__NONE = -1,
+    MPIR_HWTOPO_TYPE__NODE,
+    MPIR_HWTOPO_TYPE__PACKAGE,
+    MPIR_HWTOPO_TYPE__SOCKET,
```

Comment:
I am not clear on the usage of `NODE`, `PACKAGE`, and `SOCKET`.

Reply:
Package = Socket, Node is a memory coherent collection of many CPUs